src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/DocumentBuilder.cs

```diff
@@ -0,0 +1,533 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.DotNet.MSIdentity.Tool;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.MSIdentity.CodeReaderWriter
+{
+    internal class DocumentBuilder
+    {
+        private DocumentEditor _documentEditor;
+        private CodeFile _codeFile;
+        private CompilationUnitSyntax _docRoot;
+        private IConsoleLogger _consoleLogger;
+        internal static char[] CodeSnippetTrimChars = new char[] { ' ', '\r', '\n', ';' };
+
+        public DocumentBuilder(
+            DocumentEditor  documentEditor,
+            CodeFile codeFile,
+            IConsoleLogger consoleLogger)
+        {
+            _documentEditor = documentEditor ?? throw new ArgumentNullException(nameof(_documentEditor));
+            _codeFile = codeFile ?? throw new ArgumentNullException(nameof(codeFile));
+            _consoleLogger = consoleLogger ?? throw new ArgumentNullException(nameof(consoleLogger));
+            _docRoot = (CompilationUnitSyntax)_documentEditor.OriginalRoot ?? throw new ArgumentNullException(nameof(_documentEditor.OriginalRoot));
+        }
+
+        public void AddUsings()
+        {
+            //adding usings
+            var usingNodes = CreateUsings(_codeFile.Usings);
+            if (usingNodes.Any() && _docRoot.Usings.Count == 0)
+            {
+                _documentEditor.ReplaceNode(_docRoot, _docRoot.WithUsings(new SyntaxList<UsingDirectiveSyntax>(usingNodes)));
+            }
+            else
+            {
+                foreach (var usingNode in usingNodes)
+                {
+                    //if usings exist (very likely), add it after the last one.
+                    if (_docRoot.Usings.Count > 0)
+                    {
+                        var usingName = usingNode.Name.ToString();
+                        if (!_docRoot.Usings.Any(node => node.Name.ToString().Equals(usingName)))
+                        {
+                            _documentEditor.InsertAfter(_docRoot.Usings.Last(), usingNode);
+                        }
+                    }
+                }
+            }
+        }
+
+        //Add class members to the top of the class.
+        public ClassDeclarationSyntax AddProperties(ClassDeclarationSyntax classDeclarationSyntax)
+        { 
+            var modifiedClassDeclarationSyntax = classDeclarationSyntax;
+            if (_codeFile.ClassProperties != null && _codeFile.ClassProperties.Any() && classDeclarationSyntax != null)
```

Comment:
>(_codeFile.ClassProperties != null && _codeFile.ClassProperties.Any() [](http://example.com/codeflow?start=15&length=69)

does this mean if class is empty , i.e. no properties it would not add your new properties?